src/Xunit.Combinatorial/CombinatorialRandomAttribute.cs

```diff
@@ -0,0 +1,126 @@
+            : this(DefaultCount, DefaultMinValue, DefaultMaxValue, NoSeed)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CombinatorialRandomAttribute"/> class.
+        /// </summary>
+        /// <param name="count">
+        /// The quantity of values to generate.
+        /// Cannot be less than 1, which would conceptually result in zero test cases.
+        /// </param>
+        public CombinatorialRandomAttribute(int count)
+            : this(count, DefaultMinValue, DefaultMaxValue, NoSeed)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CombinatorialRandomAttribute"/> class.
+        /// </summary>
+        /// <param name="count">
+        /// The quantity of values to generate.
+        /// Cannot be less than 1, which would conceptually result in zero test cases.
+        /// </param>
+        /// <param name="maxValue">
+        /// maxValue for System.Random.Next method.
+        /// </param>
+        public CombinatorialRandomAttribute(int count, int maxValue)
+            : this(count, DefaultMinValue, maxValue, NoSeed)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CombinatorialRandomAttribute"/> class.
+        /// </summary>
+        /// <param name="count">
+        /// The quantity of values to generate.
+        /// Cannot be less than 1, which would conceptually result in zero test cases.
+        /// </param>
+        /// <param name="minValue">
+        /// minValue for System.Random.Next method.
+        /// </param>
+        /// <param name="maxValue">
+        /// maxValue for System.Random.Next method.
+        /// </param>
+        public CombinatorialRandomAttribute(int count, int minValue, int maxValue)
+            : this(count, minValue, maxValue, NoSeed)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CombinatorialRandomAttribute"/> class.
+        /// </summary>
+        /// <param name="count">
+        /// The quantity of values to generate.
+        /// Cannot be less than 1, which would conceptually result in zero test cases.
+        /// </param>
+        /// <param name="minValue">
+        /// minValue for System.Random.Next method.
+        /// </param>
+        /// <param name="maxValue">
+        /// maxValue for System.Random.Next method.
+        /// </param>
+        /// <param name="seed">
+        /// Seed for System.Random constructor. If Seed equal to NoSeed value then System.Random constructor whithout seed used.
+        /// </param>
+        public CombinatorialRandomAttribute(int count, int minValue, int maxValue, int seed)
+        {
+            if (count < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
+
+            var random = seed == NoSeed ? new Random() : new Random(seed);
```

Comment:
For reproducible failures it's very important to log the random number seed so it can be reproduced later. I wonder if the data-provider has an opportunity to get at the `ITestOutputHelper` from xunit to log it.
If it can log it, I am more willing to merge this PR. Otherwise I feel that this is encouraging a pattern where tests will randomly fail without a log explaining why or how to repro it.

Reply:
I don't know what log you are talking about. But Test Explorer shows values for all tests that it runned. And you can re-run or re-debug the method you need with the selected parameter values.

![Test Explorer](https://user-images.githubusercontent.com/6407980/131239974-2eee8016-a3d4-46eb-84ca-a3a8ded1e956.png)

You can also assign seed value. In this case, random values will always be the same.